src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -325,6 +331,120 @@ EnsureCitusTableCanBeCreated(Oid relationOid)
 }
 
 
+/*
+ * undistribute_table gets a distributed table name and
+ * udistributes the table.
+ */
+Datum
+undistribute_table(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+
+	Oid relationId = PG_GETARG_OID(0);
+
+	EnsureTableOwner(relationId);
+	Relation relation = relation_open(relationId, ExclusiveLock);
+
+	const char *relationName = get_rel_name(relationId);
+
+	Oid schemaId = get_rel_namespace(relationId);
+	const char *schemaName = get_namespace_name(schemaId);
+
+
+	if (TableReferencing(relationId))
+	{
+		ereport(ERROR, (errmsg("Cannot undistribute table "
+							   "because it has a foreign key.")));
+	}
+
+	if (TableReferenced(relationId))
+	{
+		ereport(ERROR, (errmsg("Cannot undistribute table "
+							   "because a foreign key references to it.")));
+	}
+
+	StringInfo query = makeStringInfo();
+
+	SPI_connect();
+
+	List *tableCommands = GetTableDDLEventsExceptCreation(relationId);
+	List *createTableCommands = GetTableCreationCommands(relationId, true);
+
+
+	StringInfo tempName = makeStringInfo();
+	CreateTemporaryTableName(&tempName, relationName);
+
+	char *createTableQuery = NULL;
+	foreach_ptr(createTableQuery, createTableCommands)
+	{
+		Node *parseTree = ParseTreeNode(createTableQuery);
+
+		if (nodeTag(parseTree) == T_CreateStmt)
+		{
+			((CreateStmt *) parseTree)->relation->relname = tempName->data;
```

Comment:
What if we use the worker_apply_ddl_command that we usually use to create shards, to create something like `mytable_0` on the coordinator?